### ISSUE TITLE
WeTek Play error on boot since specific nightly build

### ISSUE THREAD

**Issue by lookshe** [libretro/Lakka#463]
Regarding to the discussion in #282 I open a new issue.

What I have done so far:

Compiled and tested image till commit 4b7146df62fe7308bf3866b39e53818a780f9f38 without changes.

Following commit dde401dd3c29ea2be12e44e0f836e127bbabd2cb does not compile, so I merged necessary changes from 6429dc7225fcb16bd3471ae1fe29c94e16346629 to make it compile again.
> modified:       projects/WeTek_Play/linux/linux.arm.conf
> deleted:       projects/WeTek_Play/patches/linux/000-svn_rev.patch
> deleted:       projects/WeTek_Play/patches/linux/010-revert_default_dvb_core_code.patch
> deleted:       projects/WeTek_Play/patches/linux/070-remove-amvideocap-spam.patch
> deleted:       projects/WeTek_Play/patches/linux/100-fix-23hzpixelclock.patch
> deleted:       projects/WeTek_Play/patches/linux/120-enable_bluetooth.patch

After flashing the image, the same error occures like described in #282 
> sh: can't access tty; job control turned off

**Comment by Kivutar** [comment]
Thanks,

Do you remember why the WeTek Play image doesn't compile after dde401d ? What was the broken package and the error message at that point?

Can you try to add 02744500d4d0b52498fafb88283f0e3b3da4b865 too?

What bugs me is that `sh: can't access tty; job control turned off` usually happen when the system fails to mount the second partition. So it could be something in the kernel config that have to do with filesystems.

Can you try to get a bootlog?

**Comment by lookshe** [comment]
IIRC there were problems while executing the deleted patch-files. That's the reason why I want to try to get the patches working.

I will add 02744500d4d0b52498fafb88283f0e3b3da4b865 afterwards.

Are there any other posibilities to get a boot log? Because I cannot find my RS-232 cable...

**Comment by Kivutar** [comment]
We should ask on wetek forums if there are differences between the wetek play hardware.

And also, you could try the latest OpenELEC, the kernel config and the patches are directly copied from there. Maybe they fixed it since then.

**Comment by lookshe** [comment]
I compiled https://github.com/OpenELEC/OpenELEC.tv/commit/79e5aec6c97cbd40126cf17c933ea2f621ab2e80 which firstly uses the same kernel that Lakka uses and got the same error while booting. The patches from OpenELEC which sligthly differs did not seem to make a difference.
As you said, I am going to compile latest OpenELEC and afterwards I will take a look at the wetek forums and ask them for help.

**Comment by lookshe** [comment]
Latest OpenELEC give me the same error.

I opened a new discussion in wetek forum: http://www.wetekforums.com/v/index.php?p=/discussion/28166/

Will try to get a boot log as soon as the USB-to-RS232 adapter arrives.

**Comment by Kivutar** [comment]
Thank you. That's somehow lucky that the bug also happens on OE. Since they have official WeTek support, they may know what is the cause.

**Comment by Kivutar** [comment]
Any news from this?

**Comment by lookshe** [comment]
They said the problem is my old NAND installation, which is not able to boot newer kernel from SD-card. But this confuses me, because it happens since a specific kernel and not in generell with a newer kernel. I will try to confirm it with a boot log, as soon as possible. I think there were also some changes in the kernel config or somewhere else that led to this behaviour and not only because it is a newer kernel.

**Comment by Kivutar** [comment]
Is it better with the new NAND bootloader? The bootloader can be incompatible with some kernels?

**Comment by lookshe** [comment]
I didn't received my USB-to-RS232 adapter yet, so I was not able to get a boot log, which should tell me, that the kernel is incompatible, which still sounds a bit strange to me.
Even though I do not want to update my NAND installation, I will try to do this after I received the adapter and made the boot log.
Sorry that it takes so long, but I am also annoyed, that the adapter did not arrive after two weeks...

**Comment by lookshe** [comment]
Finally I was able to make a boot log, after I had to recover my WeTek. I installed OpenELEC 5.0.8 on NAND to get the same error again. After taking the boot log I installed 6.0.3 because I had to reinstall everything, so I used a newer version. With this I was able to boot new OpenELEC images from SD card, but still have problems booting Lakka. I will investigate that in the next days, but first the boot log of NAND 5.0.8 trying to boot latest Lakka from SD card: http://www.thehappy.de/~lookshe/pub/wetek_lakka.log

**Comment by Kivutar** [comment]
Can you try this new image based on LibreELEC? http://static.kivutar.me/Lakka-WeTek_Play.arm-8.0-devel-20160907223943-r23730-g012092c.img.gz

**Comment by lookshe** [comment]
This image is working booting on the first try :)
But I have some problems with the video output:
![image](http://picn.de/uploads/2016/09/i282by4c8a.jpg)

**Comment by Raxone** [comment]
Same problem like lookshe say.Problem with video output.

**Comment by Kivutar** [comment]
Can you boot on Android and change the resolution to 1080p?

**Comment by Raxone** [comment]
u-boot is last android dualboot 1.1.7 but i not use android i got enigma2
on nand.
version Lakka-WeTek_Play.arm-devel-20160109152500-r21276-g4dadcab.img
workin with old u-boot without dualboot(change boot option with remote) and
openelec or enigma2 on nand but with last u-boot dualboot not turn on hdmi
must change source on tv and than work.

**Comment by Raxone** [comment]
If install android and set 1080p dualboot not work don't boot from sdcard no matters what is on sdcard enigma2,lakka or openelec.If erase nand and try boot lakka same problem with video output.
I can easy turn on ssh but i don't know what i need change to try fix problem.

**Comment by Raxone** [comment]
I extract from Lakka-WeTek_Play.arm-devel-20160109152500-r21276-g4dadcab.img from kernel initramfs script "platform_init" and replace it in last lakka kernel build initramfs and rebuild kernel now work with all version.
I use enigma2/Openatv on nand,last u-boot (dualboot) and lakka on sdcard.

**Comment by Kivutar** [comment]
So our current platform_init script is broken?

**Comment by Raxone** [comment]
For me only change platform_init  fix problem. 
I try 2 days all possible solution but nothing fix problem.
Everething else work ok.

**Comment by Kivutar** [comment]
OK, I'm trying the platform_init from LibreELEC.

**Comment by Kivutar** [comment]
Could you try this please? http://static.kivutar.me/Lakka-WeTek_Play.arm-devel-20161017145155-r21617-g0d2198b.img.gz

**Comment by Kivutar** [comment]
@lookshe could you give it a try please?

**Comment by Raxone** [comment]
O sorry i try and got same error.

**Comment by lookshe** [comment]
Sorry for the late response. I get the same result like before :(

**Comment by Kivutar** [comment]
What about the new LE based images? http://le.builds.lakka.tv/

**Comment by Raxone** [comment]
Same video output error again on wetekplay 1.

**Comment by Kivutar** [comment]
So LibreELEC is incompatible with wetek play 1? Can you try LibreELEC 8.0 on it please?

**Comment by Raxone** [comment]
I try last image
Lakka-WeTek_Play.arm-8.0-devel-20170305041324-r25834-g76ebcbd.img and error
is same.

**Comment by Kivutar** [comment]
Thanks for your patience. If you have time, can you test this one please? http://releases.libreelec.tv/LibreELEC-WeTek_Play.arm-8.0.0.img.gz

**Comment by Raxone** [comment]
This image work.

**Comment by Kivutar** [comment]
@Ntemis, any idea of why it LE 8.0 works and not Lakka-LE 8.0? The codebase is almost 100% the same.

**Comment by Ntemis** [comment]
Yes lakka needs to force a resolution from platform init or setenv to it from bootloader

**Comment by Kivutar** [comment]
The same kind of things codesnake pushed on Hub and Play 2?

**Comment by Ntemis** [comment]
Yes but better set at 720p vs 1080p

**Comment by Kivutar** [comment]
Why?

**Comment by NuisMons** [comment]
problem still present, only `fbset -fb /dev/fb0 -g 1280 720 1280 1440 32` in ssh show normal video
can`t set fullhd
my wetekplay:
WeTekPlay Openelec edition with
libreelec 8.0.1 on nand

**Comment by Ntemis** [comment]
Can you move this issue to lakka libreelec please?
